File: xiaosa/js/APPV2.js
/**
 * 传参 ?type=url&params=http://122.228.85.203:1000@泽少1
 * 传参 ?type=url&params=http://122.228.85.203:1000@泽少2
 */

var rule = {
    title: 'APPV2[模板]',
    author: '道长',
    version: '20241012 beta1',
    update_info: `
20241012:
1.根据群友嗷呜的appv2模板修改成可传参源，类似采集之王用法传参
`.trim(),
    host: '',
    url: '/api.php/app/video?tid=fyclassfyfilter&limit=20&pg=fypage',
    filter_url: '',
    filter: {},
    homeUrl: '/api.php/app/index_video',
    detailUrl: '/api.php/app/video_detail?id=fyid',
    searchUrl: '/api.php/app/search?text=**&pg=fypage',
    parseUrl: '',
    searchable: 2,
    quickSearch: 1,
    filterable: 1,
    headers: {
        'User-Agent': 'okhttp/4.1.0'
    },
    params: 'http://122.228.85.203:1000$http://122.228.85.203:1000/play?url=',
    hostJs: $js.toString(() => {
        HOST = rule.params.split('$')[0];
    }),
    预处理: $js.toString(() => {
        log(`传入参数:${rule.params}`);
        let _host = rule.params.split('$')[0];
        rule.parseUrl = rule.params.split('$')[1];
        let _url = _host.rstrip('/') + '/api.php/app/nav?token';
        let _headers = {'User-Agent': 'Dart/2.14 (dart:io)'};
        let html = request(_url, {headers: _headers});
        let data = JSON.parse(html);
        let _classes = [];
        let _filter = {};
        let _filter_url = '';
        let dy = {"class": "类型", "area": "地区", "lang": "语言", "year": "年份", "letter": "字母", "by": "排序"};
        let jsonData = data.list;
        for (let k = 0; k < jsonData.length; k++) {
            let hasNonEmptyField = false;
            let _obj = {
                type_name: jsonData[k].type_name,
                type_id: jsonData[k].type_id,
            };
            _classes.push(_obj);
            for (let key in dy) {
                if (key in jsonData[k].type_extend && jsonData[k].type_extend[key].trim() !== "") {
                    hasNonEmptyField = true;
                    break
                }
            }
            if (hasNonEmptyField) {
                _filter[String(jsonData[k].type_id)] = [];
                for (let dkey in jsonData[k].type_extend) {
                    if (dkey in dy && jsonData[k].type_extend[dkey].trim() !== "") {
                        if (k === 0) {
                            _filter_url += `&${dkey}={{fl.${dkey}}}`
                        }
                        let values = jsonData[k].type_extend[dkey].split(',');
                        let valueArray = values.map(value => ({"n": value.trim(), "v": value.trim()}));
                        _filter[String(jsonData[k].type_id)].push({"key": dkey, "name": dy[dkey], "value": valueArray})
                    }
                }
            }
        }
        rule.classes = _classes;
        rule.filter = _filter;
        rule.filter_url = _filter_url;
    }),
    class_parse: $js.toString(() => {
        input = rule.classes;
    }),
    play_parse: true,
    lazy: $js.toString(() => {
        if (!/^http/.test(input)) {
            input = rule.parseUrl + input
        } else {
            input = {
                url: input,
                parse: 0,
                header: ''
            }
        }

    }),
    推荐: $js.toString(() => {
        let data = JSON.parse(request(input)).list;
        let com = [];
        data.forEach(item => {
            if (Array.isArray(item.vlist) && item.vlist.length !== 0) {
                com = com.concat(item.vlist)
            }
        })
        VODS = com
    }),
    一级: $js.toString(() => {
        VODS = JSON.parse(request(input)).list
    }),
    二级: $js.toString(() => {
        VOD = JSON.parse(request(input)).data
    }),
    搜索: '*',
}

File: xiaosa/py/APPV2.py
# -*- coding: utf-8 -*-
# by @嗷呜
import sys
sys.path.append('..')
from base.spider import Spider

class Spider(Spider):

    def init(self, extend=""):
        '''
        example:
        {
            "key": "py_appV2",
            "name": "xxx",
            "type": 3,
            "searchable": 1,
            "quickSearch": 1,
            "filterable": 1,
            "api": "./py/APPV2.py",
            "ext": "http://cmsyt.lyyytv.cn"
        }
        
        '''
        self.host=extend
        pass

    def getName(self):
        pass

    def isVideoFormat(self, url):
        pass

    def manualVideoCheck(self):
        pass

    def destroy(self):
        pass

    headers = {
        'User-Agent': 'okhttp/4.12.0',
    }

    def homeContent(self, filter):
        data = self.fetch(f"{self.host}//api.php/app/nav?token=",headers=self.headers).json()
        keys = ["class", "area", "lang", "year", "letter", "by", "sort"]
        filters = {}
        classes = []
        for item in data['list']:
            has_non_empty_field = False
            jsontype_extend = item["type_extend"]
            classes.append({"type_name": item["type_name"], "type_id": item["type_id"]})
            for key in keys:
                if key in jsontype_extend and jsontype_extend[key].strip() != "":
                    has_non_empty_field = True
                    break
            if has_non_empty_field:
                filters[str(item["type_id"])] = []
            for dkey in jsontype_extend:
                if dkey in keys and jsontype_extend[dkey].strip() != "":
                    values = jsontype_extend[dkey].split(",")
                    value_array = [{"n": value.strip(), "v": value.strip()} for value in values if
                                   value.strip() != ""]
                    filters[str(item["type_id"])].append({"key": dkey, "name": dkey, "value": value_array})
        result = {}
        result["class"] = classes
        result["filters"] = filters
        return result

    def homeVideoContent(self):
        data=self.fetch(f"{self.host}/api.php/app/index_video?token=",headers=self.headers).json()
        videos=[]
        for item in data['list']:videos.extend(item['vlist'])
        return {'list':videos}

    def categoryContent(self, tid, pg, filter, extend):
        params = {'tid':tid,'class':extend.get('class',''),'area':extend.get('area',''),'lang':extend.get('lang',''),'year':extend.get('year',''),'limit':'18','pg':pg}
        data=self.fetch(f"{self.host}/api.php/app/video",params=params,headers=self.headers).json()
        return data

    def detailContent(self, ids):
        data=self.fetch(f"{self.host}/api.php/app/video_detail?id={ids[0]}",headers=self.headers).json()
        return  {'list':[data['data']]}

    def searchContent(self, key, quick, pg="1"):
        data=self.fetch(f"{self.host}/api.php/app/search?text={key}&pg={pg}",headers=self.headers).json()
        videos=data['list']
        for item in data['list']:
            item.pop('type', None)
        return {'list':videos,'page':pg}

    def playerContent(self, flag, id, vipFlags):
        return  {'jx':1,'playUrl':'','parse': 1, 'url': id, 'header': self.headers}

    def localProxy(self, param):
        pass


